DashWallet/AppDelegate.m

```diff
@@ -291,6 +292,27 @@ - (void)setupDashWalletComponentsWithOptions:(NSDictionary *)launchOptions {
 
     self.balanceNotifier = [[DWBalanceNotifier alloc] init];
     [self.balanceNotifier setupNotifications];
+
+    UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
+    center.delegate = self;
+}
+
+- (void)userNotificationCenter:(UNUserNotificationCenter *)center
+       willPresentNotification:(UNNotification *)notification
+         withCompletionHandler:(void (^)(UNNotificationPresentationOptions options))completionHandler {
+    
+    if ([notification.request.identifier isEqual: CrowdNodeObjcWrapper.notificationID]) {
+        completionHandler(UNNotificationPresentationOptionAlert | UNNotificationPresentationOptionSound);
+    }
+}
+
+- (void)userNotificationCenter:(UNUserNotificationCenter *)center didReceiveNotificationResponse:(UNNotificationResponse *)response withCompletionHandler:(void(^)(void))completionHandler {
+
+    if ([response.notification.request.identifier isEqual: CrowdNodeObjcWrapper.notificationID]) {
+        UIViewController *vc = [CrowdNodeModelObjcWrapper getRootVC];
+        [_window.rootViewController presentViewController:vc animated:YES completion:nil];
+        completionHandler();
+    }
```

Comment:
We need to show notifications in the foreground as well, provided that the user isn't on the CrowdNode screen.

This code can probably go into a separate class in the future. We will have a bunch of additional logic to track crowdnode transactions and show notifications (like withdrawals, confirmation transactions, etc).